app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -109,21 +118,41 @@ public CommentsViewModel(final @NonNull Environment environment) {
 
     final Observable<List<Comment>> comments = paginator.paginatedData().share();
 
+    final Observable<Boolean> currentUserIsCreator = Observable.combineLatest(
+      currentUser,
+      project.map(Project::creator),
+      Pair::create
+    )
+      .map(userAndCreator -> userAndCreator.first != null && userAndCreator.first.id() == userAndCreator.second.id());
+
+    final Observable<Boolean> userCanComment = Observable.combineLatest(
+      currentUserIsCreator,
```

Comment:
since `currentUserIsCreator` is only used in this observable could we reduce the number of `combineLatest`s used and do all of that work in this one observable?

I think we are safe cause you gate this observables by a `takeWhen` later, but just good practice to not overuse `combineLatest`.